library/src/main/java/com/novoda/downloadmanager/DownloadBatch.java

```diff
@@ -258,8 +258,8 @@ private static boolean networkError(InternalDownloadBatchStatus downloadBatchSta
         if (status == WAITING_FOR_NETWORK) {
             return true;
         } else if (status == ERROR) {
-            DownloadError.Type downloadErrorType = downloadBatchStatus.getDownloadErrorType();
-            if (downloadErrorType == DownloadError.Type.NETWORK_ERROR_CANNOT_DOWNLOAD_FILE) {
+            DownloadError downloadError = downloadBatchStatus.downloadError();
+            if (downloadError != null && downloadError.type() == DownloadError.Type.NETWORK_ERROR_CANNOT_DOWNLOAD_FILE) {
```

Comment:
It can be inlined to:
`return DownloadError.Type.NETWORK_ERROR_CANNOT_DOWNLOAD_FILE == downloadBatchStatus.downloadError()`

Reply:
Very true 😄